src/inputosm.cpp

```diff
@@ -58,12 +58,12 @@ bool input_file(const char* filename,
 
     size_t len = strlen(filename);
     namespace fs = std::filesystem;
-    const char* extension = fs::path(filename).extension().c_str();
-    if (0 == strcasecmp(extension, ".osm") or 0 == strcasecmp(extension, ".osc"))
+    const std::string extension = fs::path(filename).extension().string();
```

Comment:
hm, I'm thinking we can maybe use the `path.compare` method: https://en.cppreference.com/w/cpp/filesystem/path/compare

since fs::path(filename).extension() already creates a path we can then use:

```c++
const fs::path extension = fs::path(filename).extension();
constexpr std::string_view k_osm = ".osm";
constexpr std::string_view k_osc = ".osc";
constexpr std::string_view k_pbf = ".pbf";
if (extension == k_osm || extension == k_osc)
...
else if (extension == k_pbf)
...
```

Reply:
we could also avoid the whole fs::path all together because it seems .extension just uses the rightmost period, so basically we can do something like:

```c++
assert(filename);
std::string_view filename_sv = filename; // does the strlen 
size_t pos_of_period = filename_sv.find_last_of('.');
std::string_view extension;
if (pos_of_period != std::string_view::npos)
{
    extesion = filename_sv.substr(pos_of_period);
}
```

Reply:
yeah, good idea!